components/prize_pool/commons/prize_pool_import.lua

```diff
@@ -542,18 +543,40 @@ function Import._groupLastVsAdditionalData(lpdbEntry)
 		table.insert(matchConditions, '[[match2id::' .. matchId .. ']]')
 	end
 
-	local matchData = mw.ext.LiquipediaDB.lpdb('match2', {
-		conditions = '[[opponent::' .. opponentName .. ']] AND (' .. table.concat(matchConditions, ' OR ') .. ')',
-		order = 'date desc, match2id desc',
-		query = 'date, match2opponents, winner',
-		limit = 1
-	})
+	local conditions = table.concat(matchConditions, ' OR ')
 
-	if not type(matchData) == 'table' or not matchData[1] then
+	if conditions ~= _last_vs_group_cache.conditions then
+		_last_vs_group_cache.conditions = conditions
+		Import._lastVsMatchesDataToCache(mw.ext.LiquipediaDB.lpdb('match2', {
+			conditions = conditions,
+			order = 'date desc, match2id desc',
+			query = 'date, match2opponents, winner',
+			limit = 1000,
+		}))
+	end
+
+	local matchData = _last_vs_group_cache.data[opponentName]
+
+	if not matchData then
 		return {}
 	end
 
-	return Import._makeAdditionalDataFromMatch(opponentName, matchData[1])
+	return Import._makeAdditionalDataFromMatch(opponentName, matchData)
+end
+
+function Import._lastVsMatchesDataToCache(queryData)
+	local byOpponent = {}
+
+	for _, match in ipairs(queryData) do
+		for _, opponent in pairs(match.match2opponents) do
+			local name = opponent.name
+			if not byOpponent[name] then
```

Comment:
Querying lastvsmatch for one opponent does not necessarily mean it is also the last match of another opponent in the same match, could this lead to issues here? (Partially related to the other comment)

Reply:
well since we are running through them from latest to earliest and only set the match as lastVs match if there is no match for that opponent already set it should work